src/data/apigateway/mod.rs

```diff
@@ -0,0 +1,170 @@
+    pub is_base64_encoded: bool,
+    pub path: String,
+    #[serde(rename="pathParameters")]
+    pub path_parameters: Option<HashMap<String, String>>,
+    #[serde(rename="queryStringParameters")]
+    pub query_string_parameters: Option<HashMap<String, String>>,
+    pub resource: String,
+    #[serde(rename="requestContext")]
+    pub request_context: HttpEventRequestContext,
+    #[serde(rename="stageVariables")]
+    pub stage_variables: Option<HashMap<String, String>>,
+}
+
+impl HttpEvent {
+
+    pub fn get_header(&self, key: &str) -> Option<&str> {
+        match self.headers {
+            Some(ref h) => h.get(key).map(|s| s.as_str()),
+            None => None,
+        }
+    }
+}
+
+impl fmt::Display for HttpEvent {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{protocol} {method} {path}{querystring} (body? {body})",
+            protocol = match self.request_context.protocol {
+                Some(ref s) => s.as_str(),
+                None => "(unknown)"
+            },
+            method = self.http_method,
+            path = self.path,
+            querystring = qs::to_string(&self.query_string_parameters).unwrap_or(String::new()),
+            body = match self.body {
+                Some(_) => true,
+                None => false
+            }
+        )
+    }
+}
+
+#[derive(Serialize,Deserialize)]
+pub struct HttpEventRequestContext {
+    #[serde(rename="accountId")]
+    pub account_id: String,
+    #[serde(rename="apiId")]
+    pub api_id: String,
+    #[serde(rename="httpMethod")]
+    pub http_method: HttpMethod,
+    pub identity: HashMap<String, Option<String>>,
+    pub path: String,
+    pub protocol: Option<String>,
+    #[serde(rename="requestId")]
+    pub request_id: String,
+    #[serde(rename="requestTime")]
+    pub request_time: Option<String>,
+    #[serde(rename="requestTimeEpoch")]
+    pub request_time_epoch: Option<u64>,
+    #[serde(rename="resourceId")]
+    pub resource_id: String,
+    #[serde(rename="resourcePath")]
+    pub resource_path: String,
+    pub stage: String,
+}
+
+impl HttpEventRequestContext {
+
+    pub fn time(&self) -> Option<DateTime<Utc>> {
+        // Utc::datetime_from_str(&self.request_time, "").ok()e
+        None
+    }
+}
+
+serde_aux_enum_number_declare!(HttpStatus {
+    OK = 200,
+
+    MovedPermanently = 301,
+    Found = 302,
+    TemporaryRedirect = 307,
+    PermanentRedirect = 308,
+
+    BadRequest = 400,
+    Unauthorized = 401,
+    Forbidden = 403,
+    NotFound = 404,
+    Gone = 410,
+
+    InternalServerError = 500,
+    BadGateway = 502,
+});
+
+#[derive(Serialize,Deserialize)]
+pub struct HttpResponse {
+    #[serde(rename="statusCode")]
+    pub status: HttpStatus,
+    pub headers: HashMap<String, String>,
+    pub body: Option<String>,
+    #[serde(rename="isBase64Encoded",default)]
+    pub is_base64: bool,
+}
+
+impl HttpResponse {
+
+    pub fn empty(status: HttpStatus) -> Self {
```

Comment:
I'm trying to convince myself to drop an existing external effort to add types to crowbar. Something I'd be missing here is a [Default impl](https://github.com/softprops/lando/blob/57cf8b4d5884c0a5e5035a316dadfa1dc596c160/src/response.rs#L29-L33)

Reply:
:+1: